packages/js/product-editor/src/components/editor/editor.tsx

```diff
@@ -41,14 +42,21 @@ import { ValidationProvider } from '../../contexts/validation-context';
 
 export type ProductEditorSettings = Partial<
 	EditorSettings & EditorBlockListSettings
->;
+> & {
+	templates: Record< string, Template[] >;
+};
 
 type EditorProps = {
-	product: Product;
+	product: Product | ProductVariation;
+	postType?: 'product' | 'product_variation';
```

Comment:
Hmm infer would only work if we pass the specific product down into `Editor` already, which I suppose we are doing for Variation. From a Types standpoint that is, we could otherwise infer it and probably get it from the post type data.
Although this will not work when we support other product types, as we won't be creating a specific page for each one. Meaning that in order.

> What happens if there is a mismatch between product and postType?

This will mean that the right product won't be viewed, or it will retrieve another one from another endpoint.
Basically the `postType` and `postId` are the primary ids. So the redux store will looking under `postType > postId`. So if the post type is mismatched it will not find the right product.

Reply:
The responsibility to render the right object is of the page and not of the editor, so we have to restrict which product should be handle by the page is being rendered. Also getting the object associated template should be load from the page depending on the object that is going to be edited and then passed to the editor as well as the object itself.